code-challenges.test.js

```diff
@@ -14,39 +14,189 @@
+// the argument is saying take the array index value, which is a string, use the string method toLowerCase and check if it includes the given letter(string), if it does return the original string value to new array, if not leave it out. 
+// return
+//
+
+// Function
+
+const fruitLetter = (array, string) => {
+	return array.filter((value) => value.toLowerCase().includes(string))
+}
+
+// Passed
+
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("twoThree", () => {
+	it("takes in an array of 5 numbers and determines whether or not the array is a 'full house'", () => {
+		const hand1 = [5, 5, 5, 3, 3]
+		// Expected output: true
+		const hand2 = [5, 5, 3, 3, 4]
+		// Expected output: false
+		const hand3 = [5, 5, 5, 5, 4]
+		// Expected output: false
+		const hand4 = [7, 2, 7, 2, 7]
+		// Expected output: true
+		expect(twoThree(hand1)).toEqual(true)
+		expect(twoThree(hand2)).toEqual(false)
+		expect(twoThree(hand3)).toEqual(false)
+		expect(twoThree(hand4)).toEqual(true)
+	})
+})
+
+// Test Failed:
+//   ● twoThree › takes in an array of 5 numbers and determines whether or not the array is a 'full house'
+// ReferenceError: twoThree is not defined
 
 // b) Create the function that makes the test pass.
+
+// Pseudo
+
+// Input: array
+// Output: boolean value true or false
+// Expect: true, false, false, true
+
+// Process
+
+// Declare a function called twoThree
+// Set parameter to array
+// Declare a new variable of cards
+// Set cards equal to array.sort(), so that all like elements are next to each other
+// Use If else statement
+// If card[0] === card[1] && card[2] === card[4] return true
+// else if card[0] === card[2] && card[3] === card[4] return true
+// else return false
+
+const twoThree = (array) => {
+	const cards = array.sort((a, b) => a - b)
+	if (
+    cards[0] === cards[1] && 
+    cards[2] === cards[3] && 
+    cards[2] === cards[4]
+  ) {
+		return true
+	} else if (
+		cards[0] === cards[1] &&
+		cards[0] === cards[2] &&
+		cards[3] === cards[4]
+	) {
+		return true
+	} else {
+		return false
+	}
+}
+
+// Passed
+
+// I originaly used the logic if (cards[0] === cards[1] && cards[2] === cards[4]) and else if (cards[0] === cards[2] && cards[3] === cards[4]), but it was brought to my attention that this was assuming the value for one of the cards, and it might fail on an edge case when there was a hand of unique card values.
+
+// Firt attempt
+
+// const twoThree = (array) => {
+//   array.sort
```

Comment:
One thing to note is that `.sort()` needs parentheses after it. This is probably why you couldn't get this approach working, because the array was not actually being sorted!

Reply:
Man! I feel like when I transition from JavaScript to Ruby, the syntax get blurry. I will have to keep working on keeping the two separate. I will try going back through this approach to see if I can get it to work!